### PR TITLE
Document the static spacing scale and generally improve the spacing guidance

### LINKED ISSUE #1210: Not clear what the 'static' spacing scale is.
The [spacing page](https://design-system.service.gov.uk/styles/spacing/#spacing-on-custom-components) refers to the 'static' spacing scale, but it's not clear what this is. From testing I think it equates to the desktop column of the responsive scale - but it's not possible to know this from the content.

I think either:
* Update the responsive table to specify which column is the static also
* Add a second table for the static scale.

### PR DESCRIPTION
**Pull request by 36degrees**
Document the static spacing scale as part of the spacing guidance, as well as:

- improving the header structure of the page
- moving the section on overrides last, to try and encourage users who are writing their own CSS to use the helpers instead
- tweaking the guidance on helpers and overrides

Suggest reviewing commit-by-commit as some of the content has moved around – this will make it easier to see what's changed.

This does **not** include documenting the new static spacing override classes (#2244) – that will be done in a follow up PR. This means that these changes can go out before v4.3.0 is released.

Fixes #1210.

### CODE REVIEW

**Comment by 36degrees** [comment]
Built on top of #2274 which should be merged first. Once that PR is merged, the target branch for this PR should automatically update to `main`.

**Comment by vanitabarrett** [comment]
This looks good from a technical point of view, and makes sense to me 👍

File: src/styles/spacing/index.md.njk [hunk 1]
```diff
@@ -123,20 +159,42 @@ For example, to apply spacing unit 6 for a 30px bottom margin on large screens a
...
+
+The spacing override classes start with `govuk-!-` followed by either `margin-` or `padding-`, and then the spacing unit you want to apply.
+
+To apply spacing in a single direction, include `left-`, `right-`, `top-`, or `bottom-` between `margin-` or `padding-` and the spacing unit.
+
+For example:
+
+- `govuk-!-margin-9` will apply a 40px margin to all sides of the element on small screens, increasing to 60px on large screens
+- `govuk-!-padding-right-5` will apply 15px of padding to the right side of the element on small screens, increasing to 25px on large screens
+- `govuk-!-margin-0` will remove all margins at all screen sizes
```
Comment (calvin-lau-sig7) [inline src/styles/spacing/index.md.njk:194]: Making an attempt to make this a bit clearer.

```suggestion
The spacing override classes start with: `govuk-!-`, followed by either `margin-` or `padding-`, and then a spacing unit number.

To apply spacing in a single direction, include `left-`, `right-`, `top-`, or `bottom-` just before the spacing unit.

For example:

- `govuk-!-margin-9` will apply a 40px margin to all sides of the element on small screens, increasing to 60px on large screens
- `govuk-!-padding-right-5` will apply 15px of padding to the right side of the element on small screens, increasing to 25px on large screens
- `govuk-!-margin-0` will remove all margins at all screen sizes
```

File: src/styles/spacing/index.md.njk [hunk 2]
```diff
@@ -77,35 +81,67 @@ The Design System uses a responsive spacing scale which adapts based on screen s
...
-The spacing override classes are structured to allow you to apply any size of the scale, using margin or padding in any direction.
-
-For example, `govuk-!-margin-9` will set a margin of 60px on large screens and 40px on small screens.
-
-To add padding use `-padding` instead of `-margin`.
+## Static spacing
 
-If you want to add the margin or padding in a particular direction, add `left` for left, `right` for right, `top` for top, or `bottom` for bottom. For example, `-margin-top` will set margin-top, `-padding-right` will set padding-right.
-If you do not specify a direction, the margin or padding will be applied to all sides of the element.
+The static spacing scale is the same as the spacing for 'large screens' in the responsive spacing scale, but always uses the same spacing regardless of the screen width.
```
Comment (calvin-lau-sig7) [inline src/styles/spacing/index.md.njk:86]: Might be better to flip this, and match the way 'Responsive' is explained?

```suggestion
The static spacing scale stays the same for all screen sizes.

It uses the same spacing as 'large screens' in the responsive spacing scale, except it does not change based on screen width.
```

Comment (36degrees) [reply src/styles/spacing/index.md.njk:86]: It feels like we're repeating ourselves a bit with 'stays the same for all screen sizes' and 'except it does not change based on screen width'

How about

```suggestion
The static spacing scale stays the same for all screen sizes, and uses the same spacing as 'large screens' in the responsive spacing scale.
```

**Pull request merged**